File: WebApi/Interfaces/ITrLineService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrLineDto, CreateTrLineDto, UpdateTrLineDto } from '../Models/TrLineDtos';

export interface ITrLineService {
  getAll(): Promise<ApiResponse<TrLineDto[]>>;
  getById(id: number): Promise<ApiResponse<TrLineDto>>;
  getByHeaderId(headerId: number): Promise<ApiResponse<TrLineDto[]>>;
  getByStockCode(stockCode: string): Promise<ApiResponse<TrLineDto[]>>;
  getByErpOrderNo(erpOrderNo: string): Promise<ApiResponse<TrLineDto[]>>;
  getActive(): Promise<ApiResponse<TrLineDto[]>>;
  getByQuantityRange(minQuantity: number, maxQuantity: number): Promise<ApiResponse<TrLineDto[]>>;
  create(createDto: CreateTrLineDto): Promise<ApiResponse<TrLineDto>>;
  update(id: number, updateDto: UpdateTrLineDto): Promise<ApiResponse<TrLineDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}

File: Interfaces/ITrLineService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrLineService
    {
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetAllAsync();
        Task<ApiResponse<TrLineDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetByHeaderIdAsync(long headerId);
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetByStockCodeAsync(string stockCode);
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetByErpOrderNoAsync(string erpOrderNo);
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<TrLineDto>>> GetByQuantityRangeAsync(decimal minQuantity, decimal maxQuantity);
        Task<ApiResponse<TrLineDto>> CreateAsync(CreateTrLineDto createDto);
        Task<ApiResponse<TrLineDto>> UpdateAsync(long id, UpdateTrLineDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}